js/XAPI.js

```diff
@@ -1118,23 +1115,41 @@ class XAPI extends Backbone.Model {
     }
 
     // Update the locally held state.
+    this.changedCollectionNames.push(collectionName);
```

Comment:
The reason I used an object rather than an array, is that with an array and your submit button example, would you have multiple 'component' entries, whereas with a `changed[collectionName] = true` and `Object.keys(changed)`, you get a unique list of changed items.

You only need to send the state update to the server once for each type.